python/scripts/archive_repositories.py

```diff
@@ -66,46 +45,25 @@ def get_environment_variables() -> tuple[str, str]:
     return github_token, github_organization_name
 
 
-def get_config_for_organization(github_organization_name: str) -> tuple[str, list[str], str] | ValueError:
+def get_config_for_organization(github_organization_name: str) -> tuple[datetime, str, list[str], str] | ValueError:
+    last_active_cutoff_date = datetime.now() - relativedelta(days=0, months=6, years=1)
+
     if github_organization_name == MINISTRYOFJUSTICE_GITHUB_ORGANIZATION_NAME:
-        return MINISTRYOFJUSTICE_GITHUB_ORGANIZATION_NAME, MINISTRYOFJUSTICE_REPOS_ALLOW_LIST, MINISTRYOFJUSTICE_REPOS_TYPE_TO_CHECK
+        return last_active_cutoff_date, MINISTRYOFJUSTICE_GITHUB_ORGANIZATION_NAME, MINISTRYOFJUSTICE_REPOS_ALLOW_LIST, MINISTRYOFJUSTICE_REPOS_TYPE_TO_CHECK
 
     if github_organization_name == MOJ_ANALYTICAL_SERVICES_GITHUB_ORGANIZATION_NAME:
-        return MOJ_ANALYTICAL_SERVICES_GITHUB_ORGANIZATION_NAME, MOJ_ANALYTICAL_SERVICES_REPOS_ALLOW_LIST, MOJ_ANALYTICAL_SERVICES_REPOS_TYPE_TO_CHECK
+        return last_active_cutoff_date, MOJ_ANALYTICAL_SERVICES_GITHUB_ORGANIZATION_NAME, MOJ_ANALYTICAL_SERVICES_REPOS_ALLOW_LIST, MOJ_ANALYTICAL_SERVICES_REPOS_TYPE_TO_CHECK
 
     raise ValueError(
         f"Unsupported Github Organization Name [{github_organization_name}]")
 
 
-def archive_inactive_repositories_by_date_and_type(github_token: str, organization_name: str, archive_date: datetime,
-                                                   allow_list: list[str],
-                                                   repo_type_to_archive: str):
-    moj_gh = MojGithub(org=organization_name, org_token=github_token)
-
-    repos = [
-        repo
-        for repo in moj_gh.get_unarchived_repos(repo_type_to_archive)
-        if ready_for_archiving(repo, archive_date)
-    ]
-
-    logging.info(
-        f"Beginning archive of inactive repositories for GitHub organization: {organization_name}")
-    logging.info("-----------------------------")
-    logging.info(
-        f"Searching for inactive repositories from date: {archive_date}")
-    logging.info("-----------------------------")
-
-    for repo in repos:
-        MojArchive(repo, allow_list).archive()
-
-
 def main():
     github_token, github_organization_name = get_environment_variables()
-    organization_name, allow_list, repo_type_to_archive = get_config_for_organization(
+    last_active_cutoff_date, organization_name, allow_list, repository_type_to_archive = get_config_for_organization(
```

Comment:
This function will cry like a baby if you don't handle that `ValueError`.

Reply:
On line 48, you say:
```
def get_config_for_organization(github_organization_name: str) -> tuple[datetime, str, list[str], str] | ValueError:
```
Should you handle the case where a `ValueError` is returned?

Reply:
That's the intended behaviour 😁

Essentially, the `get_config_for_organization()` function is saying, "if you give bad input, I will attempt to exit the program"

Then the caller (the `main()` function) doesn't handle the exception, saying, "if you think there's a problem, there must be a problem!"

It's only applicable to catch exceptions if there is an alternative flow for the application to take, given that exception. An example is the RateLimitting decorator, in those cases we want to catch that and retry after a set amount of time. Altering the flow of the application due to exceptional circumstances

But typically, exceptions are exceptional, and you usually want everything to stop if we hit one 🔥

Reply:
👆Just to add, there's also cases where you want to try/catch/finally/raise - just to ensure tidy up process are always executed before the process actually bubbles the error up 😁

Typically seen with database transactions, don't want to write that someone's account has been credited if the debit fails for the other account 💸